### DISCUSSION TITLE
Connecting to project resources from containers.

### DISCUSSION THREAD

**Discussion by darren-clark** [microsoft/aspire#1808] [Q&A] [answered]
I think this is a Docker networking issue, but one that I can't work around without additional Aspire configuration options.

My problem is I need to launch a container, that then acts as a gateway/proxy for multiple services. In this case it's an Apollo Router if that makes a difference. Currently we just use the Apollo container image, with some minor changes for certificates and configuration. I would really rather not have to run the router outside of the container, and in fact I'm pretty sure that wouldn't help since it would need to run in WSL.

So what's happening is that I can't connect back from the container to the projects running in Windows launched by Aspire, even when using `host.docker.internal`. I'm pretty sure the issue is that the proxy endpoints are listening on 127.0.0.1 and not 0.0.0.0, or the Docker host interface.

Any ideas?

I guess my other option is to get Aspire to launch `rover dev` which is what I've been doing manually up until this point.

**Comment by davidfowl** [comment]
I'm not sure how to help without deeply understanding your network and machine configuration, so I'll offer this https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/networking-overview. Let us know if it helps.

**Comment by e-master** [comment]
i have a very similar problem, and at first i attributed it to using podman instead of docker desktop, but now i think it's just simply not supported by Aspire.

my setup looks like this:
![image](https://github.com/dotnet/aspire/assets/13695064/ff9f5e24-8b42-46f5-9f4b-e5c1856fe866)

I started another docker container using Aspire that needs to connect to RabbitMQ but it can't. I am able to ping `host.containers.internal` from the third party container, but there's nothing bound to the 60502 port. 

As @darren-clark , i also think that it's mostly because Aspire binds by default to 127.0.0.1 instead of the default 0.0.0.0, but i'm not an expert in networking by any means, so perhaps i missed something:
![image](https://github.com/dotnet/aspire/assets/13695064/d0428d13-2574-4d6e-bc32-a19a9c48d985)

**Comment by e-master** [comment, marked as answer]
Turns out it was a local podman networking issue!
I still don't know how to resolve it, but after reading this https://github.com/dotnet/aspire/discussions/2677 i tried to switch from Podman to Rancher and enable the _WSL -> Enable networking tunnel_ option, and now the scenario on the diagram above works fine.
